core/EventDispatcher.php

```diff
@@ -78,29 +80,36 @@ public function postEvent($eventName, $params, $pending = false, $plugins = null
             $this->pendingEvents[] = array($eventName, $params);
         }
 
+        $manager = $this->getPluginManager();
+
         if (empty($plugins)) {
-            $plugins = $this->getPluginManager()->getPluginsLoadedAndActivated();
+            $plugins = $manager->getPluginsLoadedAndActivated();
         }
 
         $callbacks = array();
 
         // collect all callbacks to execute
-        foreach ($plugins as $plugin) {
-            if (is_string($plugin)) {
-                $plugin = $this->getPluginManager()->getLoadedPlugin($plugin);
+        foreach ($plugins as $pluginName) {
```

Comment:
When inserting in bulk this can be easily executed a 100k times. Those changes don't make `postEvent` blazing fast but faster.